lib/dora/explorer.ex

```diff
@@ -43,6 +43,7 @@ defmodule Dora.Explorer do
   def handle_info(:request_logs, state) do
     events =
       HttpRpc.events(state.address, state.last_block)
+      |> Enum.filter(&(not &1["removed"]))
```

Comment:
Aren't we missing an `Enum.uniq`? What happens if we get back two events with `removed: false`?

Reply:
I tried it with just this `filter` and it seems to be enough to make it work, apparently. I'm now working on adding a constraint to the database table and having it take care of that part. With the link you sent me, I noticed that using `block_hash` + `tx_hash` + `log_index` we should be able to ensure unique logs. A `tx_hash` is always unique by design, but it can trigger more than one event per tx. That is when we'll look into `log_index`.

>// LogIndex is the index of the event log in the sequence of events produced by the message execution.
	// (this is the index in the events AMT on the message receipt)
	
This means that if a Log/Event has the same index in the event array for the same transaction, it means that they are actually the same log!

Does it make sense?

Reply:
From that description of the LogIndex, yes, I'd say it make perfect sense